server/plat/rtdial.cpp

```diff
@@ -471,6 +471,13 @@ int gdial_os_application_resume(const char *app_name, int instance_id) {
     return GDIAL_APP_ERROR_NONE;
 }
 
+int gdial_os_application_state_wait(const char *app_name, int instance_id, GDialAppState state, unsigned int timeout_ms) {
+    const char * str_state = state == GDIAL_APP_STATE_RUNNING ? "running" :
+                              state == GDIAL_APP_STATE_HIDE ? "suspended" :
+                              state == GDIAL_APP_STATE_STOPPED ? "stopped" : "?";
```

Comment:
maybe we should return immediately in "?" case? (it will always timeout anyway ...)